k8s.io

```diff
@@ -0,0 +1 @@
+common-protos/k8s.io/
```

Comment:
can we avoid polluting the top level repo with these symlinks?

Seems to work if we do this:

```diff
diff --git a/k8s.io b/k8s.io
deleted file mode 120000
index bfbd5e7da3..0000000000
--- a/k8s.io
+++ /dev/null
@@ -1 +0,0 @@
-common-protos/k8s.io/
\ No newline at end of file
diff --git a/tools/proto/buf.yaml b/tools/proto/buf.yaml
index 2e48bffcbc..cb5757c683 100644
--- a/tools/proto/buf.yaml
+++ b/tools/proto/buf.yaml
@@ -1,6 +1,8 @@
-version: v1
+version: v1beta1
 build:
-  excludes:
+  roots:
+    - operator
+    - pkg
     - common-protos
 lint:
   allow_comment_ignores: true
diff --git a/tools/proto/proto.mk b/tools/proto/proto.mk
index b8455e7b84..2e7431cafd 100644
--- a/tools/proto/proto.mk
+++ b/tools/proto/proto.mk
@@ -19,7 +19,7 @@ BUF_CONFIG_DIR := tools/proto
 proto: operator-proto dns-proto echo-proto workload-proto zds-proto
 
 operator-proto:
-	buf generate --config $(BUF_CONFIG_DIR)/buf.yaml --path operator/pkg/ --template $(BUF_CONFIG_DIR)/buf.golang.yaml
+	buf generate --config tools/proto/buf.yaml --path operator/pkg/ --output operator --template tools/proto/buf.golang.yaml
 
 dns-proto:
 	buf generate --config $(BUF_CONFIG_DIR)/buf.yaml --path pkg/dns/ --template $(BUF_CONFIG_DIR)/buf.golang.yaml
```

Reply:
I noticed that it is structured this way for i[stio/api](https://github.com/istio/api/blob/master/k8s.io) repository but I agree that we should avoid polluting the top level repo.
I can revert these changes

Reply:
Yeah I set it up originally for istio/api and wasn't happy with that either but I couldn't figure out an alternative.. at least at the time, maybe a similar approach would work there.

Reply:
I think in api they need to construct all protos in one `buf` call is why actually

Reply:
Revert done